src/core/mesh/qgsmeshlayer.h

```diff
@@ -377,6 +377,23 @@ class CORE_EXPORT QgsMeshLayer : public QgsMapLayer
       */
     void setReferenceTime( const QDateTime &referenceTime );
 
+    /**
+      * Returns the position of the snapped point on the closest mesh element
```

Comment:
if I am reading the code correctly, when there is an error (e.g. triangular mesh not intialized), it return invalid Point. good to add to the docs .